### DISCUSSION TITLE
how to pass values from checkbox, dropdown selection to multiple steps form

### DISCUSSION THREAD

**Discussion by swimhiking** [react-hook-form/react-hook-form#2836] [Q&A]
I found pass values by useContext can only pass values from input box; If I have dropdown list or checkbox, then I cannot pass values by useContext hook to next step form. such as first example work, second not work

case 1
<input
          type="text"
          name="address"
          placeholder="Enter your address"
          defaultValue={profile.address1}
          ref={register({ required: true})}
          />
case 2
<select name="state"
     defaultValue={profile.state}    
        placeholder="Choose..."  
       ref={register}
         >
        <option>Choose...</option>
        <option>Alberta</option>
        <option>British Columnbia</option>
        <option>Ontario</option>
        <option>...</option>
</select>

How to make the second case also work? Thank you.

**Comment by bluebill1049** [comment]
provide more details with a codesandbox.

**Reply by swimhiking** [reply]
https://codesandbox.io/s/holy-cache-y8bji?file=/src/App.js

In the example, not sure why line#9 not work; the point is I have 2 dom elements input box and checkbox: input box value can be pass to other component, while checkbox, dropdown list won't work. this just an example. Thanks

**Reply by bluebill1049** [reply]
That's not related to hook form.

**Reply by bluebill1049** [reply]
<img width="1273" alt="Screen Shot 2020-09-08 at 12 33 01 pm" src="https://user-images.githubusercontent.com/10513364/92427277-7152d680-f1cf-11ea-9ba8-97041eba3f67.png">
looking inot your own custom hook: useMultiFormContext

**Comment by swimhiking** [comment]
Yes, it's working fine in my local or deploy as a node application, not
sure why it has error in codesandbox

**Reply by bluebill1049** [reply]
perhaps you want to reproduce this issue in a simple and isolated env, which doesn't need that extra logic (makes others life easier as well)

**Comment by swimhiking** [comment]
Yes, sure, you remind me something, I will make it testable tomorrow. Thanks

**Comment by swimhiking** [comment]
Hi Bill,

I have the working version in here:
https://codesandbox.io/s/holy-cache-y8bji?file=/src/Profile.js

You can put some string in a zip input box and click the checkbox to pass
a boolean value; input string value is passed to the next Review page,
checkbox cannot. This is what I found that only input box values can share
cross pages, others checkbox, dropdown cannot. Let me know what's the issue
it has, or Hook form need to add these features? I believe checkbox,
dropdowns are all normal elements in a form. Thanks

Allen

**Comment by swimhiking** [comment]
By the way, If I have more than 1 input form pages, for example I can apply
a required constraint to an input box at second page, but when I try to go
back to first page, it shouldn't have this constrain; if I try to go to
next third page, then this constraint has to apply. I found this is also
difficult to implement. Thanks

Allen

**Reply by bluebill1049** [reply]
That's at your implementation detail. I would recommend you to watch this video tutorial: 

https://www.youtube.com/watch?v=evDxlqnsxXc
https://www.youtube.com/watch?v=CeAkxVwsyMU

**Comment by swimhiking** [comment]
Thanks Bill,

The video is good, I guess I will try to use little-state-machine to
implement.

Allen

**Reply by bluebill1049** [reply]
no worries. there is a codesandbox as well, you can fork it and have a play.
